File: www/cgi-bin/hrima/post/form/post.php
<?php

header('Content-Type: text/html');

$msg = "<html><body>";
$msg .= "<h1>PHP response</h1>";
if (isset($_POST['name']) && isset($_POST['age'])) {
    header("HTTP/1.1 200 OK");
    $name = htmlspecialchars($_POST['name']);
    $age = htmlspecialchars($_POST['age']);
    $msg .= "<p>User has name: " . $name . "</p>";
    $msg .= "<p>And age: " . $age . "</p>";
} else {
    header("HTTP/1.1 400 NOK");
    $msg .= "<h2>Name and age are required!</h2>";
}

$msg .= "</body></html>";

header("Content-Length: " . strlen($msg));

echo $msg;

?>

File: www/cgi-bin/hrima/post/form/post.py
#!/usr/bin/python3

import cgi

form = cgi.FieldStorage()

msg = "<html><body>"
msg += "<h1>Python response</h1>\n"
if "name" and "age" in form:
    header = "HTTP/1.1 200 OK"
    name = form.getvalue("name")
    age = form.getvalue("age")
    msg += "<p>user has name: " + name + "</p>"
    msg += "<p>and age: " + age + "</p>"
else:
    header = "HTTP/1.1 400 NOK"
    msg += "<h2>name, age, are required!</h2>"
msg += "</body></html>"


print("Content-type:text/html")
print("Content-Length:", len(msg), "")
print("")
print(msg)